spec/integration/assign_spec.rb

```diff
@@ -54,6 +57,47 @@ def assert_publish_topic_state(topic, user)
         UserArchivedMessage.move_to_inbox!(user.id, pm.reload)
       end
     end
+
+    it 'publishes the right message on archive and move to inbox for groups' do
+      assigner = TopicAssigner.new(pm, user)
+      assigner.assign(group)
+
+      assert_publish_topic_state(pm, user) do
+        GroupArchivedMessage.archive!(group.id, pm.reload)
+      end
+
+      assert_publish_topic_state(pm, user) do
+        GroupArchivedMessage.move_to_inbox!(group.id, pm.reload)
+      end
+    end
+
+    it "unassign and assign user if unassign_on_group_archive" do
+      SiteSetting.unassign_on_group_archive = true
+      assigner = TopicAssigner.new(pm, user)
+      assigner.assign(user)
+
+      GroupArchivedMessage.archive!(group.id, pm.reload)
+      expect(pm.assignment).to eq(nil)
+      expect(pm.custom_fields["prev_assigned_to_id"]).to eq(user.id.to_s)
```

Comment:
Hmm did we not register this as an integer custom field? Odd that we need to do `to_s`